src/segmentedfilesystem/PacketManager.java

```diff
@@ -0,0 +1,215 @@
+package segmentedfilesystem;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+public class PacketManager {
+
+    byte[] packet;
+    List<HeaderPacket> headerList;
+    List<DataPacket> dataList;
+
+    public PacketManager(){
+        this.headerList = new ArrayList<HeaderPacket>(); //list for the header packets
+        this.dataList = new ArrayList<DataPacket>(); //list for all the data packets
+    }
+
+    public boolean isHeader(byte status) { // determines if a packet is a header packet or not
+        boolean header = false;
+        if (status % 2 == 0) {
+            header = true;
+        } 
+        return header;
+    }
+
+    /*
+    This method will take the array of bytes and calculated length and create either a dataPacket or headerPacket. 
+    It uses the isHeader method as a way to determine if the given array of bytes is either a dataPacket or headerPacket
+    */
+    public void createPack(byte[] aPacket, int length) throws IOException { 
+        if (isHeader(aPacket[0]) == true) {  
+            newHeader(aPacket, length);
+        }
+        if (isHeader(aPacket[0]) != true) {
+            newData(aPacket, length);
+        }
+        if (theEnd() == true){
+            System.out.println(dataList.size());
+            Map<Byte, List <DataPacket>> sortedList = sortDataList(dataList);
+            Byte[] uniqueFileIDs = sortedList.keySet().toArray(new Byte[0]);
+            List<DataPacket> data1 = sortedList.get(uniqueFileIDs[0]);
```

Comment:
This is really pretty ugly, TBH, and really error prone. You're definitely not using the cool features of a `Map` well here. I'd be happy to talk about how you might clean this up, but honestly I'm having trouble figuring out what you're trying to do here.